lib/SIL/Linker.cpp

```diff
@@ -70,10 +85,7 @@ bool SILLinkerVisitor::linkInVTable(ClassDecl *D) {
   // for processing.
   bool Result = false;
   for (auto P : Vtbl->getEntries()) {
-    if (P.Implementation->isExternalDeclaration()) {
-      Result = true;
-      addFunctionToWorklist(P.Implementation);
-    }
+    Result |= maybeAddFunctionToWorklist(P.Implementation);
```

Comment:
I haven't thought about this (I am asking you to do so), but are you sure that having a uniform condition across all of these is safe. For instance just off the top of my head, imagine if we could specialize witness tables. Would the linkage be shared? If we then serialized that, wouldn't we have shared external linkage and then wouldn't deserialize the function?

Also given that you are changing the conditions, can you add tests that verify this new behavior for other parts of this patch?

Reply:
Safe is the wrong word here, I meant are you sure that expanding the conditions here what we want to do.

Reply:
hasSharedVisibility() returns true for both shared and shared external. Also note that linkInVTable() is only called it isLinkAll() is true, so in this case the condition is equivalent to F->isExternalDefinition(). So there’s no real functional change here. I’ll add an assert and comments.